SDK Showcase/SDK Showcase/Views/Categories/CategoryView.swift

```diff
@@ -5,47 +5,85 @@ import SwiftUI
         HStack {
-            List {
-                Text(category.description)
-                    .multilineTextAlignment(.leading)
-                    .font(.system(size: 15))
-                    .foregroundStyle(.secondary)
-                
-                Section(header: Text("Required Actions")) {
-                    ForEach(category.requiredActions, id:\.self) { action in
-                        ActionView(action: binding(for: action))
+            ZStack {
+                List {
+                    Text(category.description)
+                        .multilineTextAlignment(.leading)
+                        .font(.system(size: 15))
+                        .foregroundStyle(.secondary)
+                    
+                    if !category.requiredActions.isEmpty {
+                        Section(header: Text("Required Actions")) {
+                            ForEach(category.requiredActions, id:\.self) { action in
+                                ActionView(action: binding(for: action))
+                            }
+                        }
+                    }
+                    
+                    if !category.optionalActions.isEmpty {
+                        Section(header: Text("Optional Actions")) {
+                            ForEach(category.optionalActions, id:\.self) { action in
+                                ActionView(action: binding(for: action))
+                            }
+                        }
+                    }
+                    
+                    if !category.selection.isEmpty {
+                        Section(header: Text("Select")) {
+                            ForEach(category.selection, id:\.self) { selection in
+                                Button(action: {
+                                    buttonAction(for: selection)
+                                }, label: {
+                                    HStack {
+                                        if let logo = selection.logo {
+                                            Image(systemName: logo)
+                                        }
+                                        Text(selection.name)
+                                    }
+                                })
+                            }
+                            
+                        }
                     }
+                    
+                    Section(content: {
+                        if isProcessing {
+                            ProgressView()
+                                .progressViewStyle(CircularProgressViewStyle())
+                        } else {
+                            TextResult(result: system.isSDKInitialized ? "SDK initialized" : "SDK not initialized \(errorValue)")
+                            TextResult(result: system.isRegistered ? "User registered as \(userData.userId ?? "")" : "User not registered")
+                        }
+                    }, header: {
+                        Text("Result")
```

Comment:
"Result"? What result?

Reply:
Result of the action

Reply:
Too generic to me. That should be, I don't know: "success", "failure", "try again" etc.

Reply:
hahaha this is a header's description

Reply:
Now it makes sense 😄